cmd/vela-slack/main.go

```diff
@@ -377,3 +422,72 @@ func run(c *cli.Context) error {
 	// execute the plugin
 	return p.Exec()
 }
+
+// Retrieves sAMAccountName from LDAP server using build author's email.
+func getSAMAccountName(c *cli.Context) string {
+
+	// LDAP environment variables
+	email := c.String("build-author-email")
+	username := c.String("ldap-username")
+	password := c.String("ldap-password")
+	ldapServer := c.String("ldap-server")
+	ldapPort := c.String("ldap-port")
+	ldapSearchBase := c.String("ldap-search-base")
+
+	// return if LDAP info not provided
+	if username == "" || password == "" {
+		return ""
+	}
+
+	// create LDAP client
+	roots := x509.NewCertPool()
+	caCerts, err := ioutil.ReadFile(c.String("sslcert.path"))
+	if err != nil {
+		logrus.Errorf("%s", err)
+		return ""
+	}
+	roots.AppendCertsFromPEM(caCerts)
+
+	config := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+		ServerName: ldapServer,
+		RootCAs:    roots,
+	}
+
+	l, err := ldap.DialTLS("tcp", fmt.Sprintf("%s:%s", ldapServer, ldapPort), config)
+	if err != nil {
+		logrus.Errorf("%s", err)
+		return ""
+	}
+
+	err = l.Bind(username, password)
+	if err != nil {
+		logrus.Errorf("%s", err)
+		return ""
+	}
+
+	filter := fmt.Sprintf("mail=%s", email)
+	req := ldap.NewSearchRequest(
+		ldapSearchBase,
+		ldap.ScopeWholeSubtree, ldap.NeverDerefAliases, 0, 0, false,
+		fmt.Sprintf("(%s)", filter),
+		[]string{"dn", "displayName", "sAMAccountName", "mail"},
+		nil,
+	)
+
+	// search for records
+	sr, err := l.Search(req)
+	if err != nil {
+		logrus.Errorf("%s", err)
+		return ""
+	}
+
+	if len(sr.Entries) != 1 {
+		logrus.Errorf("user does not exist or too many entries returned: %d", len(sr.Entries))
+		return ""
+	}
+
+	// return sAMAccountName
+	sAMAccountName := sr.Entries[0].GetAttributeValue("sAMAccountName")
```

Comment:
just a thought here, so no action to take now, but i wonder if this should be configurable in the future to allow folks to define their own attribute in case they have a different one suits better.